In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/anime_reviews.csv")

print(df.shape)

(1490, 19)


In [2]:
df.head()

,review_id,anime_mal_id,anime_title,username,score,tags,review_text,date,episodes_watched,is_spoiler,is_preliminary,reactions_overall,reactions_nice,reactions_love_it,reactions_funny,reactions_confusing,reactions_informative,reactions_well_written,reactions_creative
0,602806,16782.0,Kotonoha no Niwa,gimmefood365,8,Recommended,Makoto Shinkai's The Garden of Words is one of...,2026-04-26,NaN,False,False,0,0,0,0,0,0,0,0
1,602801,50594.0,Suzume no Tojimari,Andrew2706,10,Recommended,Y con es termino de ver todas las peliculas de...,2026-04-26,NaN,False,False,0,0,0,0,0,0,0,0
2,602799,5680.0,K-On!,yui_keion_fan,10,Recommended,My honest opinion on K-ON!: I absolutely LOVE ...,2026-04-26,NaN,False,False,0,0,0,0,0,0,0,0
3,602793,1735.0,Naruto: Shippuuden,Kiunax3,10,Recommended,"Uh, I don't even even know where to start. Hon...",2026-04-26,NaN,False,False,0,0,0,0,0,0,0,0
4,602787,227.0,FLCL,CR7SPY,8,Recommended,"""FLCL"" both walks and works only on a thin lin...",2026-04-26,NaN,False,False,0,0,0,0,0,0,0,0


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1490 entries, 0 to 1489
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   review_id               1490 non-null   int64  
 1   anime_mal_id            50 non-null     float64
 2   anime_title             50 non-null     str    
 3   username                1490 non-null   str    
 4   score                   1490 non-null   int64  
 5   tags                    1490 non-null   str    
 6   review_text             1490 non-null   str    
 7   date                    1490 non-null   str    
 8   episodes_watched        0 non-null      float64
 9   is_spoiler              1490 non-null   bool   
 10  is_preliminary          1490 non-null   bool   
 11  reactions_overall       1490 non-null   int64  
 12  reactions_nice          1490 non-null   int64  
 13  reactions_love_it       1490 non-null   int64  
 14  reactions_funny         1490 non-null   int64  
 15

In [4]:
df.columns.tolist()

['review_id',
 'anime_mal_id',
 'anime_title',
 'username',
 'score',
 'tags',
 'review_text',
 'date',
 'episodes_watched',
 'is_spoiler',
 'is_preliminary',
 'reactions_overall',
 'reactions_nice',
 'reactions_love_it',
 'reactions_funny',
 'reactions_confusing',
 'reactions_informative',
 'reactions_well_written',
 'reactions_creative']

In [5]:
df[['anime_title', 'anime_mal_id']].head(20)

,anime_title,anime_mal_id
0,Kotonoha no Niwa,16782.0
1,Suzume no Tojimari,50594.0
2,K-On!,5680.0
3,Naruto: Shippuuden,1735.0
4,FLCL,227.0
5,Boku no Hero Academia: Sukue! Kyuujo Kunren!,33929.0
6,Fangkai Nage Nuwu,62927.0
7,"Hazure Skill ""Kinomi Master"": Skill no Mi (Tab...",59002.0
8,Love Live! Nijigasaki Gakuen School Idol Douko...,40879.0
9,Nanatsu no Maken ga Shihai suru,50582.0


In [6]:
df["review_text"].iloc[0]

'Makoto Shinkai\'s The Garden of Words is one of those films that earns the word "immersive" without having to ask for it. From the first scene, it pulls you in — not with action, not with urgency, but with the quiet insistence of rain on glass. You settle into it before you\'ve even decided to. The visuals hit first, and they hit hard. Every backdrop — wet garden tiles, light fracturing through leaves, the particular grey of a Tokyo sky about to open up — looks less like animation and more like memory. Character designs are precise without being sterile. There\'s a tactile quality to everyscene, a feeling that if you reached into the screen you\'d come back with damp fingers. It\'s the kind of craft that makes you pause mid-scene just to sit with what you\'re looking at. Knowing this came out in 2013 almost feels like a trick.\n\nThe story doesn\'t try to compete with the visuals on spectacle. It doesn\'t need to. A teenage boy and an older woman keep finding each other in the same pa

In [7]:
len(df["review_text"].iloc[0])

2000

In [8]:
df["score"].describe()

count    1490.000000
mean        7.380537
std         2.589335
min         1.000000
25%         5.000000
50%         8.000000
75%        10.000000
max        10.000000
Name: score, dtype: float64

In [9]:
df["score"].value_counts().sort_index()

score
1      41
2      31
3      84
4      95
5     132
6     120
7     138
8     165
9     244
10    440
Name: count, dtype: int64

In [10]:
df["review_text"].str.len().describe()

count    1490.000000
mean     1743.854362
std       447.955479
min       368.000000
25%      1616.250000
50%      2000.000000
75%      2000.000000
max      2000.000000
Name: review_text, dtype: float64

In [11]:
(df["review_text"].str.len() == 2000).sum()

np.int64(1017)

In [12]:
((df["review_text"].str.len() == 2000).sum() / len(df))*100

np.float64(68.25503355704699)

In [13]:
def score_to_sentiment(score):
    if score >= 8:
        return "Positive"
    elif score >= 5:
        return "Neutral"
    else:
        return "Negative"

df["sentiment"] = df["score"].apply(score_to_sentiment)

In [14]:
df[["score", "sentiment"]].head(10)

,score,sentiment
0,8,Positive
1,10,Positive
2,10,Positive
3,10,Positive
4,8,Positive
5,7,Neutral
6,10,Positive
7,1,Negative
8,8,Positive
9,9,Positive


In [15]:
df["sentiment"].value_counts()

sentiment
Positive    849
Neutral     390
Negative    251
Name: count, dtype: int64

In [16]:
df["tags"].value_counts().head(20)

tags
Recommended                          967
Mixed Feelings                       249
Not Recommended                      218
Not Recommended|Funny                 44
Mixed Feelings|Funny                   4
Recommended|Informative                2
Recommended|Funny                      2
Mixed Feelings|Well-written            2
Not Recommended|Funny|Informative      1
Not Recommended|Well-written           1
Name: count, dtype: int64

In [17]:
print(df["review_text"].iloc[0][:500])

Makoto Shinkai's The Garden of Words is one of those films that earns the word "immersive" without having to ask for it. From the first scene, it pulls you in — not with action, not with urgency, but with the quiet insistence of rain on glass. You settle into it before you've even decided to. The visuals hit first, and they hit hard. Every backdrop — wet garden tiles, light fracturing through leaves, the particular grey of a Tokyo sky about to open up — looks less like animation and more like me


In [18]:
df["review_length"] = df["review_text"].str.len()

df["review_length"].describe()

count    1490.000000
mean     1743.854362
std       447.955479
min       368.000000
25%      1616.250000
50%      2000.000000
75%      2000.000000
max      2000.000000
Name: review_length, dtype: float64

In [20]:
## Do positive reviews tend to be longer than negative reviews?
df.groupby("sentiment")["review_length"].mean()

sentiment
Negative    1733.844622
Neutral     1790.841026
Positive    1725.229682
Name: review_length, dtype: float64

# Dataset Audit Findings

## Finding 1
68.25% of reviews are exactly 2000 characters long, suggesting truncation or a platform review limit.

## Finding 2
Review length appears to have little relationship with sentiment.
Average review lengths are similar across Positive, Neutral, and Negative classes.

## Finding 3
The dataset is mildly imbalanced:
- Positive: 849
- Neutral: 390
- Negative: 251

## Finding 4
episodes_watched is completely missing and is unlikely to be useful.